### ISSUE TITLE
Serializing ManagerOptions

### ISSUE THREAD

**Issue by wireless90** [iotaledger/wallet.rs#1452]
## Bug description

I am trying to use C# to connect to wallet.rs, leveraging on the swift bindings found [here](https://github.com/iotaledger/wallet.rs/blob/bbebd26198732e768cf58ff407fbe3cdcf01c5e4/bindings/swift/src/lib.rs).

I am having trouble to serialize the `manager_options` argument in function `iota_initialize`.

It is more to do with the pascal casing vs camel casing json properties.

In the end, I had to mix and match the casings which could degrade developer experience.

## Rust version

Which version of Rust are you running?

edition = "2021"

## Version

Which version of the library are you using?

iota-wallet = { git = "https://github.com/iotaledger/wallet.rs", branch = "develop" , features = ["message_interface", "events"]}

## Hardware specification

What hardware are you using?

- Operating system: 
- RAM:
- Cores:
- Device:

## Steps To reproduce the bug

Explain how the maintainer can reproduce the bug.

The first json I sent was the default in C# which is PascalCasing.

```json
{
  "StoragePath": "./walletdb",
  "CoinType": 4219,
  "SecretManager": "{\"Stronghold\":{\"Password\":\"password\",\"SnapshotPath\":\"./jsonStronghold\"}}",
  "ClientOptions": "{\"Nodes\":[\"https://api.testnet.shimmer.network\"],\"LocalPow\":false,\"FallbackToLocalPow\":true,\"Offline\":false}"
}
```
<img width="623" alt="image" src="https://user-images.githubusercontent.com/12537739/191731687-c1323114-ceb0-4392-a50e-501c4426d5e1.png">

The error I got back was 
>{MissingParameter("client_options")}

This error message got me thinking perhaps snake case was expected.

Thus for the second try I did with snake_casing.

```json
{
  "storage_path": "./walletdb",
  "coin_type": 4219,
  "secret_manager": "{\"stronghold\":{\"password\":\"password\",\"snapshot_path\":\"./jsonstronghold\"}}",
  "client_options": "{\"nodes\":[\"https://api.testnet.shimmer.network\"],\"local_pow\":false,\"fallback_to_local_pow\":true,\"offline\":false}"
}
```

<img width="629" alt="image" src="https://user-images.githubusercontent.com/12537739/191735428-7583b641-cb76-4452-9700-3fb216056980.png">

The error I got back was the same.

>{MissingParameter("client_options")}

So its very likely the json binding problem still existed.

So for the third try, i tried camelCasing.

```json
{
  "storagePath": "./walletdb",
  "coinType": 4219,
  "secretManager": "{\"stronghold\":{\"password\":\"password\",\"snapshotPath\":\"./jsonstronghold\"}}",
  "clientOptions": "{\"nodes\":[\"https://api.testnet.shimmer.network\"],\"localPow\":false,\"fallbackToLocalPow\":true,\"offline\":false}"
}
```
Fortunately, the binding problem for `clientOptions` was solved however, a new error.

> {ClientError(Json(Error("unknown variant `stronghold`, expected one of `Stronghold`, `Mnemonic`, `HexSeed`, `Placeholder`", line: 1, column: 13)))}

Thus I had to camelCase everything but Pascalcase the `"stronghold`" property.

```json
{
  "storagePath": "./walletdb",
  "coinType": 4219,
  "secretManager": "{\"Stronghold\":{\"password\":\"password\",\"snapshotPath\":\"./jsonStronghold\"}}",
  "clientOptions": "{\"nodes\":[\"https://api.testnet.shimmer.network\"],\"localPow\":false,\"fallbackToLocalPow\":true,\"offline\":false}"
}
```

This worked.

TLDR: `clientOptions` leans towards camelCasing while `Stronghold` leans towards PascalCasing. This could damper dev experience. Different languages could have different default casings for serialization. Case-agnostic would be better.

This could be worked along #1451
